packages/api/src/services/sdkService.ts

```diff
@@ -52,15 +87,27 @@ export const evaluateFlag = async (
                     (y) => y.context === firstContextKey,
                 )[0];
                 if (matchCondition) {
-                    const hasMatch = getMatchByCondition(
-                        matchCondition,
-                        contextValue,
-                    );
-                    response.match = hasMatch;
-                    response.meta.segment = x.segment.key;
-                    response.meta.condition = matchCondition.key;
-                    foundMatchCondition = true;
-                    response.reason = ApiResponseCodes.FlagMatch;
+                    if (matchCondition.conditionType === "string") {
+                        const hasMatch = getMatchByCondition(
```

Comment:
getMatchByCondition is where the check on the conditionType should happen. please move the call of  datetimeConditionMatcher into that function.